Fixes Disease Outbreak: Advanced [NO GBP]

lessthnthree opened the pull request:
## About The Pull Request
- Corrects severity thresholds for med/harmful/dangerous
- Removes tier 1 cures from cure pool
- Dangerous/Biohazard disease correctly show up on HUD at spawn
- RNG actually listens to what severity the symptoms are supposed to be
- Fixes various symptoms that were in the wrong severity list
- Disease is now assigned illness name based on its highest severity component
- Converts old code to snake_case
## Why It's Good For The Game
Disease Outbreak works as expected, both on random spawn or admin custom disease creation. Ensures that diseases that are harmful or potential for quick spread are visible to scanners and HUDs.
## Changelog
:cl: LT3
code: Disease Outbreak: Disease max severity replaced with requested severity
code: Disease Outbreak: Generated disease now includes symptom based name
code: Disease Outbreak: Disease name is now included in the station announcement
code: Disease Outbreak: Now in snake case
balance: Disease Outbreak: Tier 1 cures removed from pool
fix: Disease Outbreak: Dangerous/Biohazard diseases are now visible
fix: Disease Outbreak: Event setup generates the correct severity
fix: Disease Outbreak: Various potential symptoms were in the wrong symptom pool
/:cl:

lessthnthree committed:
update disease list and fix severity selection

lessthnthree committed:
symptoms 1

lessthnthree committed:
symptoms 2

lessthnthree committed:
capitalization

lessthnthree commented on code/modules/events/disease_outbreak.dm:
```diff
@@ -335,26 +349,24 @@
-	var/res = rand(1, 6)
-	if(res == oldres)
-		return
+	var/res = rand(2, 6)
```
I'll change it from a removal

lessthnthree committed:
forgot to document this

lessthnthree commented on code/datums/diseases/advance/symptoms/symptoms.dm:
```diff
@@ -3,6 +3,7 @@
 /datum/symptom
 	var/name = ""
 	var/desc = "If you see this something went very wrong." //Basic symptom description
+	var/illness = "Unrecognizable" // Basic illness name
```
Ah yeah, whoops. Forgot to doc the vars.

lessthnthree committed:
add illness to announcement

lessthnthree committed:
request should be odd number severities

san7890 commented on code/modules/events/disease_outbreak.dm:
```diff
@@ -83,11 +83,13 @@
 	///The disease type we will be spawning
 	var/datum/disease/virus_type
+	///The preset (classic) or generated (advanced) illness name
+	var/datum/disease/illness_type
```
why are you casting `illness_type` if all you do is read the name string? just do `var/illness_type = ""`

lessthnthree replied:
Fixed.

san7890 commented on code/modules/events/disease_outbreak.dm:
```diff
@@ -267,25 +274,35 @@
+			current_severity = new_symptom.severity
 
 	if(prob(50))
 		symptoms += new /datum/symptom/youth
+		name = "Advanced [name]"
 
-	Refresh()
+	if(prob(50))
+		symptoms += new /datum/symptom/viralevolution
```
Shouldn't making it an advanced virus be on the prob(50) that we apply viral evolution on instead of the youth one, or do i misunderstand?

lessthnthree replied:
50% chance of getting that particular trait. It can have both as well.

san7890 commented on code/modules/events/disease_outbreak.dm:
```diff
@@ -307,25 +324,27 @@
 
-		else if(transmissibility < ADV_SPREAD_MID)
-			SetSpread(DISEASE_SPREAD_CONTACT_SKIN)
+			else if(transmissibility < ADV_SPREAD_MID)
+				SetSpread(DISEASE_SPREAD_CONTACT_SKIN)
 
-		else
-			SetSpread(DISEASE_SPREAD_AIRBORNE)
-			visibility_flags &= ~HIDDEN_SCANNER
+			else
+				SetSpread(DISEASE_SPREAD_AIRBORNE)
+				visibility_flags &= ~HIDDEN_SCANNER
```
This doesn't make sense to be embedded into the switch statement like this IMO, just have the switch statement call `GenerateCure()` and return if we match `severity` rather than have all of these new variables be introduced into the switch statement.

lessthnthree replied:
This is setting the transmissibility (spread) not the severity, based on the value the rand generated in the var.

san7890 commented on code/modules/events/disease_outbreak.dm:
```diff
@@ -267,25 +274,35 @@
+
+		var/datum/symptom/new_symptom = new chosen_symptom
+
+		if(symptoms.len == (max_symptoms - 1) && (current_severity < requested_severity) && (new_symptom.severity < requested_severity))
```
I feel like doing the whole `symptons.len == (max_symptoms - 1)` check should be at the top of this while loop, before we declare `new_symptom` variable as a type of early continue. Also, might be apt to do something like `symptoms.len >= max_symptoms`, not sure why you decrement max_symptoms by 1 for the evaluation, smells odd. Unsure though, but should incorporate some sort of early continue before we even start declaring variables.

lessthnthree replied:
You're right, it should be checking if the symptom is valid before creating it.

For the second part, symptoms.len == (max_symptoms - 1) is to continue only if this is the last symptom we're trying to pick, though while moving it I realized the potential difference in severity by adding this is miniscule so it can be removed.

san7890 reviewed (changes requested):
some things that were kinda :/ to me on the nose

lessthnthree committed:
Merge remote-tracking branch 'upstream/master' into disease-adv-2

lessthnthree committed:
illness_tyle

lessthnthree committed:
Update disease_outbreak.dm

lessthnthree committed:
various changes

removes duplicate diseases we generate below
set min/max custom virus boundaries
add early returns in various spots
lower stage speed via changing evolution to adaptation w/EY
slightly lower stage_prob

san7890 commented on code/datums/diseases/advance/advance.dm:
```diff
@@ -503,7 +503,7 @@
  *  Make virus visible to heath scanners
  */
-/datum/disease/advance/proc/MakeVisible()
+/datum/disease/advance/proc/makeVisible()
```
if you're going to rename it, make it snake case `make_visible()`.

san7890 commented on code/modules/events/disease_outbreak.dm:
```diff
@@ -85,11 +85,13 @@
 
 /datum/round_event/disease_outbreak/announce(fake)
-	priority_announce("Confirmed outbreak of level 7 viral biohazard aboard [station_name()]. All personnel must contain the outbreak.", "Biohazard Alert", ANNOUNCER_OUTBREAK7)
+	priority_announce("Confirmed outbreak of level 7 viral biohazard aboard [station_name()]. All personnel must contain the outbreak.", "[illness_type] Alert", ANNOUNCER_OUTBREAK7)
```
Is `illness_type` null-safe in this scenario? Sounds like it should be but just wanna make sure since the default is a null (more on that in another review).

lessthnthree replied:
Yes, I tested it receiving null as illness_type and it's fine. In symptoms.dm we also give it a default of  var/illness = "Unidentified" so it should always have something.

san7890 commented on code/modules/events/disease_outbreak.dm:
```diff
@@ -85,11 +85,13 @@
 	///The disease type we will be spawning
 	var/datum/disease/virus_type
+	///The preset (classic) or generated (advanced) illness name
+	var/illness_type
```
```suggestion
	var/illness_type = ""
```

Do this to "type" it as a string. Could also make it a generic name like `var/illness_type = "Biohazard"` or something, but at least make it string-typed.

san7890 commented on code/modules/events/disease_outbreak.dm:
```diff
@@ -165,22 +168,23 @@
 				return ADMIN_CANCEL_EVENT
 
-		chosen_max_symptoms = tgui_input_number(usr, "How many symptoms do you want your virus to have?", "A pox upon ye!", 4, 15)
+		chosen_max_symptoms = tgui_input_number(usr, "How many symptoms do you want your virus to have?", "A pox upon ye!", 4, 7, 1)
```
label your args here so that the next person who comes around doesn't have to read the proc documentation to figure out what number means what (i had to read the proc documentation)

san7890 commented on code/modules/events/disease_outbreak.dm:
```diff
@@ -271,25 +279,40 @@
+			name = "[new_symptom.illness]"
+			current_severity = new_symptom.severity
+
+	//Finalize the generated virus with EY on the end.
```
What's `EY`? It's not abundantly clear with the code (I presume it has something to do with the youth datum, but it's just not helpful in its current state). There's plenty of space to write out what needs to be written out without using acronyms.

lessthnthree replied:
Okay, I'll expand it.

san7890 commented on code/modules/events/disease_outbreak.dm:
```diff
@@ -301,21 +324,22 @@
+		if(severity == "Dangerous" || severity == "BIOHAZARD")
+			visibility_flags &= ~HIDDEN_SCANNER
+			SetSpread(DISEASE_SPREAD_CONTACT_SKIN)
+			return
```
We return without calling `GenerateCure()`, intended?

lessthnthree replied:
No, it should be calling GenerateCure, but not bother with the transmissibility roll and if chain

san7890 commented on code/modules/events/disease_outbreak.dm:
```diff
@@ -301,21 +324,22 @@
-	addtimer(CALLBACK(src, PROC_REF(MakeVisible)), ((ADV_ANNOUNCE_DELAY * 2) - 10) SECONDS)
+	addtimer(CALLBACK(src, PROC_REF(makeVisible)), ((ADV_ANNOUNCE_DELAY * 2) - 10) SECONDS)
 
 	if(length(properties))
```
Make this an early return, we don't need to do this indentation.

san7890 commented on code/modules/events/disease_outbreak.dm:
```diff
@@ -271,25 +279,40 @@
+
+	//Advanced variants have increased resistance but spread slower
+	if(prob(40))
+		symptoms += new /datum/symptom/viraladaptation
```
Why're we changing it from viralevolution to viraladaptation?

lessthnthree replied:
I was generating more diseases today working on the other things from the review, and wanted to slow down the disease progression a bit from what I was seeing. So swap out some speed for resistance, and the stage_prob multiplier is reduced from 0.5 to 0.4. This'll make the final product more consistent, less outliers with say, stage speed 12.

lessthnthree replied:
After sleeping on this, I shouldn't try to box in the values so much, originally I had both Adaptation and Evolution in there, and I think I'll switch it back.

lessthnthree committed:
Update code/modules/events/disease_outbreak.dm

Co-authored-by: san7890 <the@san7890.com>

lessthnthree committed:
make_visible

lessthnthree committed:
Update disease_outbreak.dm

lessthnthree committed:
Merge branch 'disease-adv-2' of https://github.com/lessthnthree/tgstation into disease-adv-2

lessthnthree committed:
generatecure

lessthnthree committed:
document args

lessthnthree committed:
convert to snek_case

lessthnthree committed:
re-add viral evolution
